x-pack/plugins/uptime/server/lib/adapters/telemetry/kibana_telemetry_adapter.ts

```diff
@@ -28,22 +28,153 @@ export class KibanaTelemetryAdapter {
-    const bucket = this.getBucketToIncrement();
-    this.collector[bucket].overview_page += 1;
+  public static countPageView(pageView: PageViewParams) {
+    const bucketId = this.getBucketToIncrement();
+    const bucket = this.collector[bucketId];
+    if (pageView.page === 'Overview') {
+      bucket.overview_page += 1;
+    }
+    if (pageView.page === 'Monitor') {
+      bucket.monitor_page += 1;
+    }
+    if (pageView.page === 'Settings') {
+      bucket.settings_page += 1;
+    }
+    this.updateDateData(pageView, bucket);
+    return bucket;
+  }
+
+  public static updateDateData(
+    { dateStart, dateEnd, autoRefreshEnabled, autorefreshInterval }: PageViewParams,
+    bucket: UptimeTelemetry
+  ) {
+    const prevDateStart = [...bucket.dateRangeStart].pop();
+    if (!prevDateStart || prevDateStart !== dateStart) {
+      bucket.dateRangeStart.push(dateStart);
+      bucket.dateRangeEnd.push(dateEnd);
+    } else {
+      const prevDateEnd = [...bucket.dateRangeEnd].pop();
+      if (!prevDateEnd || prevDateEnd !== dateEnd) {
+        bucket.dateRangeStart.push(dateStart);
+        bucket.dateRangeEnd.push(dateEnd);
+      }
+    }
+
+    const prevAutorefreshInterval = [...bucket.autorefreshInterval].pop();
+    if (!prevAutorefreshInterval || prevAutorefreshInterval !== autorefreshInterval) {
+      bucket.autorefreshInterval.push(autorefreshInterval);
+    }
+    bucket.autoRefreshEnabled = autoRefreshEnabled;
   }
 
-  public static countMonitor() {
+  public static async countNoOfUniqueMonitorAndLocations(callCluster: APICaller) {
+    const params = {
+      index: INDEX_NAMES.HEARTBEAT,
+      body: {
+        query: {
+          bool: {
+            must: [
+              {
+                range: {
+                  '@timestamp': {
+                    gte: 'now-1d/d',
+                    lt: 'now',
+                  },
+                },
+              },
+            ],
+          },
+        },
+        size: 0,
+        aggs: {
+          unique_monitors: {
+            cardinality: {
+              field: 'monitor.id',
+            },
+          },
+          unique_locations: {
+            cardinality: {
+              field: 'observer.geo.name',
+              missing: 'N/A',
+            },
+          },
+          monitor_name: {
+            string_stats: {
+              field: 'monitor.name',
+            },
+          },
+          observer_loc_name: {
+            string_stats: {
+              field: 'observer.geo.name',
+            },
+          },
+          monitors: {
+            terms: {
+              field: 'monitor.id',
+              size: 10000,
+            },
+            aggs: {
+              docs: {
+                top_hits: {
+                  size: 1,
+                  _source: ['monitor.timespan'],
+                },
+              },
+            },
+          },
+        },
+      },
+    };
+
+    const result = await callCluster('search', params);
```

Comment:
It might be nice in an ideal world to have some type validation for this response, so we avoid having all the `any` fields/vars below. I think it's ok to take this on as tech debt given the relatively small effort it would be, to avoid losing velocity with this change.